e2e/version/version.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package version
+
+import (
+	"os/exec"
+	"testing"
+
+	"github.com/blang/semver"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+
+func verifyVersion(t *testing.T) {
```

Comment:
I would break this function up into two test functions.

1. Loop around the different ways we test version and make sure they are semantic
2. Loop around the different ways we test version and make sure they are all equal.